pkgs/games/shipwright/default.nix

```diff
@@ -0,0 +1,203 @@
+      To get this file you can dump your Ocarina of Time Master Quest's cartridge to a file,
+      and add it to the nix store with nix-store --add-fixed sha256 <FILE>.
+
+      The supported variants are:
+       - debug-mq: Ocarina of Time Debug PAL GC MQ (Dungeons will be Master Quest)
+
+      This is optional if you have imported an Ocarina of Time ROM.
+      If so, please set enableOotRom to true and enableOotMqRom to false.
+      If both are checked, Ship of Harkinian will be built with both ROMs.
+      Alternatively, use the `shipwright` package instead of `shipwright-mq`.
+    '';
+    sha256 = {
+      debug-mq = "161a3e8d6e67ecd5d996eb0251a946e6ae2b65089563ab1cd0873eca609b6a5f";
+    }.${mqRomVariant};
+  }
+, enableOotRom ? true
+, enableOotMqRom ? false
+}:
+
+assert (enableOotRom || enableOotMqRom);
+
+stdenv.mkDerivation rec {
+  pname = "shipwright";
+  version = "5.1.3";
+
+  dontUseCmakeConfigure = true;
+
+  src = fetchFromGitHub {
+    owner = "harbourmasters";
+    repo = "shipwright";
+    rev = "776886925172040c8e260afba829aa5c0f448414";
+    hash = "sha256-nDjaFrlBYXgxBW4BvsEI1uJVvXHgNwsro44B2Td85bI=";
+    fetchSubmodules = true;
+  };
+  rev = src.rev;
+
+  enableParallelBuilding = true;
+
+  # Copied from the building guide
+  cmakeFlags = [
+    "DEBUG=0"
+    "OPTFLAGS=-O2"
+  ];
+
+  # Linking fails without this
+  hardeningDisable = [ "all" ];
+
+  # Many warnings are being treated as errors
+  NIX_CFLAGS_COMPILE = [
+    "-Wno-error"
+  ];
+
+  nativeBuildInputs = [ python3 llvmPackages.bintools copyDesktopItems makeWrapper cmake ninja pkg-config lsb-release clang imagemagick ];
+
+  buildInputs = [ libX11 libXrandr libXinerama libXcursor libXi libXext glew SDL2 SDL2_net libpulseaudio libpng bzip2 StormLib ];
+
+  postPatch = ''
+    # This script attempts to call git to retrieve the commit hash,
+    # so we have to manually patch it in instead.
+    substituteInPlace ZAPDTR/ZAPD/genbuildinfo.py \
+      --replace "label = subprocess" "#label = subprocess" \
+      --replace "+ label +" "+ \"${builtins.substring 0 7 rev}\" +"
+
+    # LIBULTRASHIP: Patch the localdir function to point to dir in user's home.
+    substituteInPlace libultraship/src/core/Window.cpp \
+      --replace "return \".\";" "return ((std::string) std::getenv(\"HOME\")) + \"/.local/share/shipofharkinian\";" \
+
+    # SHIPWRIGHT:  to point to dir in user's home.
+    substituteInPlace soh/soh/OTRGlobals.cpp \
+      --replace "Ship::Window::GetPathRelativeToAppDirectory(\"oot-mq.otr\");" "(\"./oot-mq.otr\");" \
+      --replace "Ship::Window::GetPathRelativeToAppDirectory(\"oot.otr\");" "(\"./oot.otr\");" \
+      --replace "Ship::Window::GetPathRelativeToAppDirectory(\"mods\");" \
+        "((std::string) std::getenv(\"HOME\")) + \"/.local/share/shipofharkinian/mods\";"
+  '';
+
+  buildPhase = ''
+    # Generate Ninja project
+    cmake -H. -Bbuild-cmake -GNinja -DCMAKE_BUILD_TYPE:STRING=Release
```

Comment:
Why is this done manually? Adding `ninja` to `nativeBuildInputs` automatically sets the `-GNinja` flag in `cmakeConfigurePhase`, it seems like there is no reason not to use that.

In addition, overriding `buildPhase` does not seem great. CMake can take care of the configuring and building (done on this line and 145 now) automatically, it does not look like we're requiring non-standard behaviour.